fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/aggregation/FilteredAggregation.java

```diff
@@ -67,12 +68,14 @@ public FilteredAggregation(@NotNull String name, String fieldName, Aggregation a
      * @return A pair where pair.left is the outer aggregation and pair.right is the inner.
      */
     @Override
-    public Pair<Aggregation, Aggregation> nest() {
-        Pair<Aggregation, Aggregation> wrappedAggNested = this.getAggregation().nest();
-        Aggregation inner = this.withAggregation(wrappedAggNested.getRight());
-        Aggregation outer = wrappedAggNested.getLeft();
+    public Pair<Optional<Aggregation>, Optional<Aggregation>> nest() {
+        Pair<Optional<Aggregation>, Optional<Aggregation>> wrappedAggNested = this.getAggregation().nest();
+        Aggregation inner = this.withAggregation(wrappedAggNested.getRight().get());
+        Aggregation outer = wrappedAggNested.getLeft().get();
```

Comment:
You definitely don't want to do it this way.
You should handle the wrapped nest being Optional.empty by making the filtered aggregation return optional empty if there's no nested agg to filter.